waspc/src/Wasp/Analyzer/Parser/PrettyPrinter.hs

```diff
@@ -0,0 +1,16 @@
+module Wasp.Analyzer.Parser.PrettyPrinter
+  ( prettyPrintParserResult,
+  )
+where
+
+import Wasp.Analyzer.Parser hiding (withCtx)
+import Wasp.Analyzer.Parser.AST.PrettyPrinter (prettyPrintAST, prettyPrintCtx)
+import Wasp.Analyzer.Parser.ParseError (getErrorMessageAndCtx)
+import Wasp.Util (indent)
+
+prettyPrintParserResult :: Either ParseError AST -> String
+prettyPrintParserResult (Left err) =
+  let (message, ctx) = getErrorMessageAndCtx err
+      locationStr = "at " ++ prettyPrintCtx ctx
+   in "Parse error " ++ locationStr ++ ":\n" ++ indent 2 message ++ "\n"
+prettyPrintParserResult (Right ast) = prettyPrintAST ast
```

Comment:
Hm, does it make sense to flip the pattern match (i.e., put the `Right` case first because it's a one liner). I realize it also makes sense to do `Left` and then `Right` because it's natural to go from left to right. Not sure which one is more idiomatic, I'll leave that to you.

Another (possibly subjective) remark, I think using a `where` block makes this a bit easier to understand (I also moved the `"at"`):
```haskell
prettyPrintParserResult (Left err) = "Parse error at" ++ locationStr ++ ":\n" ++ indent 2 message ++ "\n"
  where
    (message, ctx) = getErrorMessageAndCtx err
    locationStr = prettyPrintCtx ctx
```
And one last thing I'm never sure of: Is it appropriate to make this function responsible for the trailing newline, or is that something we should leave to the caller?

Reply:
Love the suggestion!
I think it makes sense to handle Right first since it is shoter + it is a positive / normal use case, while Left is an "error" use case, which I would prefer reading after I read the core logic.
I also added that where clause, it does make it easier to read this.
As for newline -> ha that is a tricky question, it really comes down to how we establish a convention with ourselves. I would actually vote for not having it here, so that caller can choose if they want it or not. I moved it out.

Reply:
Oh, and if you remove the newline, you may find it more readable to construct the final string using `unlines`.

Reply:
Correction: I haven't moved it out at the end as the whole pretty parser logic is providing newlines in this fashion, embedded into the thing it prints, even for lower level stuff like statements, expressions and similar, so I would have to refactor the whole thing to use this different approach of caller supplying newlines, which I think would be too much work at the moment compared to what we get for it.

Reply:
I like it like this, but I have a feeling you'd prefer to change the order of variables in the where clause :D